code/modules/mob/living/silicon/robot/robot_items.dm

```diff
@@ -1,3 +1,533 @@
+							if(reagents.has_reagent(bad_reagent))
+								badshit += reagents_to_log[bad_reagent]
+						if(badshit.len)
+							var/hl="\red <b>([english_list(badshit)])</b> \black"
+							message_admins("[user.name] ([user.ckey]) added [reagents.get_reagent_ids(1)] to \a [target] with [src].[hl] (<A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[user.x];Y=[user.y];Z=[user.z]'>JMP</a>)")
+							log_game("[user.name] ([user.ckey]) added [reagents.get_reagent_ids(1)] to \a [target] with [src].")
+
+				trans = src.reagents.trans_to(target, amount_per_transfer_from_this)
+				user << "\blue You transfer [trans] units of the solution."
+				if (src.reagents.total_volume<src.reagents.maximum_volume)
+					icon_state = "[initial(icon_state)]"
+
+			else if (1)//Draw Reagents
+
+				if(src.reagents.total_volume >= src.reagents.maximum_volume)
+					user << "\red Your ChemVac is full!"
+					return
+				if(!target.is_open_container() && !istype(target,/obj/structure/reagent_dispensers))
+					user << "\red You cannot directly remove reagents from [target]."
+					return
+
+				if(!target.reagents.total_volume)
+					user << "\red [target] is empty."
+					return
+
+				var/trans = target.reagents.trans_to(src, amount_per_transfer_from_this)
+
+				user << "\blue You fill the [src] with [trans] units of the solution."
+
+				if(src.reagents.total_volume >= src.reagents.maximum_volume)
+					icon_state = "[initial(icon_state)][1]"
+
+			return
+
+	update_icon()
+		overlays.Cut()
+		if(ismob(loc))
+			var/injoverlay
+			switch(mode)
+				if (0)
+					injoverlay = "inject"
+				if (1)
+					injoverlay = "draw"
+			overlays += injoverlay
+
+
+/obj/item/weapon/robot/medical/nanopastesynth
+	name = "Nanopaste Synthesizer"
+	desc = "Synthesizes nanites that repair damage to robotic machinery and actuate your autorepair systems. Consumes 100 charge per use."
+	icon = 'icons/obj/nanopaste.dmi'
+	icon_state = "tube"
+	var/usecost = 100
+	var/cooldown = 0
+	var/cooldown_time = 50
+
+/obj/item/weapon/robot/medical/nanopastesynth/attack(mob/living/M as mob, mob/user as mob)
+	if (!istype(M) || !istype(user))
+		return 0
+	if(cooldown && user == M)
+		playsound(user, 'sound/machines/buzz-two.ogg', 50, 0)
+		user << "<span class='notice'>Your auto-repair system is recharging.</span>"
+		return
+	var/mob/living/silicon/robot/U = user
+	if(U.cell.charge < usecost)
+		user << "<span class='notice'>Your energy levels are inadequate to use this module.</span>"
+		return
+	if (istype(M,/mob/living/silicon/robot))	//Repairing cyborgs
+		var/mob/living/silicon/robot/R = M
+		if (R.getBruteLoss() || R.getFireLoss() )
+			R.adjustBruteLoss(-15)
+			R.adjustFireLoss(-15)
+			R.updatehealth()
+			U.cell.use(usecost)//Deducts charge.
+			user.visible_message("<span class='notice'>\The [user] applied some [src] at [R]'s damaged areas.</span>",\
+				"<span class='notice'>You apply some [src] at [R]'s damaged areas.</span>")
+			//Cooldown if using it on yourself.
```

Comment:
Letting medbots repair themselves. Do I actually have to go into how bad of an idea this is?